### ISSUE TITLE
Launching VScode from the quick access menu does not correctly inherit system environment variable if a change has been made since first launch.

### ISSUE THREAD

**Issue by geoff3jones** [Microsoft/vscode#20140]
- VSCode Version: Code 1.9.0 (27240e71ef390bf2d66307e677c2a333cebf75af, 2017-02-02T08:31:00.827Z)
- OS Version: Windows_NT ia32 10.0.14393
- Extensions:

|Extension|Author|Version|
|---|---|---|
|python|donjayamanne|0.5.8|
|live-html-previewer|hdg|0.0.1|
|latex-workshop|James-Yu|0.2.13|
|cpptools|ms-vscode|0.10.0|
|latex|ms-vscode|0.1.3|
|code-spell-checker|streetsidesoftware|0.17.2|
|vim|vscodevim|0.5.2|

---

Steps to Reproduce:

1. open an instance of VScode
2. change the PATH environment variable
3. reload VScode observe inability to detect the new PATH changes.

Fixes:

a) load the integrated terminal (powershell) and reset the system the local PATH variable to that of the system i.e. [thanks stackoverflow](http://stackoverflow.com/questions/17794507/reload-the-path-in-powershell)
```
$env:Path = [System.Environment]::GetEnvironmentVariable("Path","Machine") + ";" + System.Environment]::GetEnvironmentVariable("Path","User")
```
This is only temporary - if i close all instances and then restart VScode then the path variable reverts to the out of date version... going to try Fixes b.
b) Reboot - (hypothesised as this would reseed all processes with the system envs)

Context: 

This was encountered by updating my MiKTeX installation after installing a the latex-workshop extension. I rebooted VScode and noticed that it couldn't detect any pdflatex binary in the PATH variable, loading up an external console I verified that it was accessible. So I killed all VScode windows (I had a second instance open) and reopened and was presented by the same error. Finally I opened up the integrated console in VScode and saw that it had the old PATH env variable so applied Fixes a).

Is it possible there is some persistent console or background process that is hanging on to the out of date?

**Comment by geoff3jones** [comment]
yup fixes b works

**Comment by Tyriar** [comment]
This looks like it just overrides the PATH environment in your local environment, not globally (@daviwil?).

```powershell
$env:Path = [System.Environment]::GetEnvironmentVariable("Path","Machine") + ";" + System.Environment]::GetEnvironmentVariable("Path","User")
```

If this is the case then this behavior is expected.

Here's how the environment works in VS Code and the terminal:

- All VS Code windows share the environment of the first (the main process), you need to close all windows to refresh the environment.
- The terminal environment is inherited (cloned) from the VS Code window, meaning any changes here will not affect the window, new windows or new terminals.

**Comment by daviwil** [comment]
Yep, setting $env:PATH does not change the path at the system level, it only persists for that PowerShell session.  If you want the change to be permanent you need to use [`System.Environment.SetEnvironmentVariable`](https://msdn.microsoft.com/en-us/library/system.environment.setenvironmentvariable%28v=vs.110%29.aspx?f=255&MSPPError=-2147217396).  Excerpt from [this documentation page](https://technet.microsoft.com/en-us/library/ff730964.aspx):

>Like we said, though, that approach only works for process-level environment variables. To create more permanent environment variables (i.e., user-level or machine-level) you need to use the .NET Framework and the SetEnvironmentVariable method.

**Comment by geoff3jones** [comment]
Hi folks, I appreciate that I wasn't clear in my initial report. I should have made it clear that I was changing the system PATH variable, and by including the temporary fix setting the local terminal PATH variable I muddied the water as to what the issue was, so I'll try to be a bit more specific.

Issue:
VScode is sometimes not picking up **system PATH changes** even after closing all VScode instances.

I have done some further testing and narrowed things down to the interaction with the quick access menu.
_Note for this I uninstalled all extensions_.

Recreation:
1) load a folder in VScode from the **quick access menu** folder icon (click **red** below)
![image](https://cloud.githubusercontent.com/assets/4126206/22733516/ff72f8e8-ede9-11e6-9fe7-91bcee1e5a86.png)
2) change PATH environment variable at the system level (Control panel dialouge etc...)
3) close ALL VScode instances
4) open VScode by clicking on the **taskbar icon** (click **white** above)
5) open terminal see the correct PATH variable YAY
6) close ALL VScode instances
7) open VScode by clicking on the **quick acces menu** folder icon (click **red** above)
8) open terminal see the out-of-date/incorrect PATH variable

**Comment by Tyriar** [comment]
@bpasero apparently the environment is stable when opening via the quick access menu (hover over Taskbar)?

**Comment by bpasero** [comment]
I need the tl;dr; here. 

Clicking on the taskbar is the same as running `code <folder>` from the command line.

**Comment by Tyriar** [comment]
Look at recreation section in https://github.com/Microsoft/vscode/issues/20140#issuecomment-278296230

**Comment by bpasero** [comment]
Ok, no idea what's going on, there is defniteley no difference in our code launching from taskbar vs quick menu, so I suspect an upstream issue.

**Comment by bpasero** [comment]
Btw the summary is still talking about something else: 

> Not detecting env PATH changes made after a VScode instance has been opened

**Comment by geoff3jones** [comment]
Hi guys thanks for looking at this.

@bpasero Please do go ahead with changing the summary of the bug, I'm not sure if/how to do it myself. At the time I hadn't managed to narrow it down to the quick access menu maybe something along the lines of this would be more appropriate: 

> Launching VScode from the quick access menu does not correctly inherit system environment variable if a change has been made since first launch.

**Comment by geoff3jones** [comment]
Sorry corrected it now to the suggetion

**Comment by bpasero** [comment]
Still happening?

**Comment by geoff3jones** [comment]
I've tested on three computers two of which were running older versions (1.16.1 & 1.17.0) of VS code (one x86), the issue appears to persist. I updated all three computers to the latest version 1.18.0 and tested again as per my comment a #20140 and the issue persists.

Sorry I don't have any new information, if it would be helpful I could spin up a clean windows install in a vm although I have tested across three different computers so far?
